vllm/attention/backends/dual_chunk_flash_attn.py

```diff
@@ -0,0 +1,877 @@
+                query.dtype)
+            query_inter = (query_inter * mscale.view(-1, 1, 1, 1)).to(
+                query.dtype)
+
+        outputs_list = []
+        softmax_lses_list = []
+
+        # intra-attention
+        seq_lens_intra = cache_seqlens - chunk_num_curr * chunk_len
+        max_seq_len_intra = seq_lens_intra.max().item()
+        block_table_intra = torch.zeros(
+            batch_size,
+            (max_seq_len_intra - 1) // block_size + 1,
+            dtype=block_table.dtype,
+            device=block_table.device,
+        )
+        for i in range(batch_size):
+            st = chunk_num_curr[i] * chunk_len // block_size
+            ed = min(
+                st + (max_seq_len_intra - 1) // block_size + 1,
+                (cache_seqlens[i] - 1) // block_size + 1,
+            )
+            block_table_intra[i, :ed - st] = block_table[i, st:ed]
+        intra_output, intra_softmax_lse = (
+            self._pagedattention_forward_decode_with_exp_sums(
+                query,
+                key_cache,
+                value_cache,
+                block_table_intra,
+                seq_lens_intra,
+                softmax_scale,
+                alibi_slopes,
+                causal=False,
+            ))
+        outputs_list.append(intra_output)
+        softmax_lses_list.append(intra_softmax_lse)
+
+        # succ-attention
+        seq_lens_succ = (chunk_num_curr -
+                         (chunk_num_curr - 1).clip(min=0)) * chunk_len
+        max_seq_len_succ = seq_lens_succ.max().item()
+        if max_seq_len_succ:
+            block_table_succ = torch.zeros(
+                batch_size,
+                (max_seq_len_succ - 1) // block_size + 1,
+                dtype=block_table.dtype,
+                device=block_table.device,
+            )
+            for i in range(batch_size):
+                st = ((chunk_num_curr[i] - 1).clip(min=0) * chunk_len //
+                      block_size)
+                ed = min(
+                    st + (max_seq_len_succ - 1) // block_size + 1,
+                    (cache_seqlens[i] - 1) // block_size + 1,
+                )
+                block_table_succ[i, :ed - st] = block_table[i, st:ed]
+            succ_output, succ_softmax_lse = (
+                self._pagedattention_forward_decode_with_exp_sums(
+                    query_succ,
+                    key_cache,
+                    value_cache,
+                    block_table_succ,
+                    seq_lens_succ,
+                    softmax_scale,
+                    alibi_slopes,
+                    causal=False,
+                ))
+            outputs_list.append(succ_output)
+            softmax_lses_list.append(succ_softmax_lse)
+
+        # inter-attention
+        seq_lens_inter = (chunk_num_curr - 1).clip(min=0) * chunk_len
+        max_seq_len_inter = seq_lens_inter.max().item()
+        if max_seq_len_inter:
+            inter_output, succ_softmax_lse = (
```

Comment:
should be `inter_softmax_lse` instead of `succ_softmax_lse`

Reply:
Thanks for pointing out the typo. Will revise in the next commit.